tests/e2e/modelregistry_test.go

```diff
@@ -0,0 +1,294 @@
+		// must be the latest one
+		t.Run("Validate Disabling ModelRegistry Component", componentCtx.validateModelRegistryDisabled)
+	})
+}
+
+func (mr *ModelRegistryTestCtx) validateModelRegistryInstance(t *testing.T) {
+	g := mr.WithT(t)
+
+	g.Eventually(
+		mr.List(gvk.ModelRegistry),
+	).Should(And(
+		HaveLen(1),
+		HaveEach(And(
+			jq.Match(`.metadata.ownerReferences[0].kind == "%s"`, gvk.DataScienceCluster.Kind),
+			jq.Match(`.spec.registriesNamespace == "%s"`, mr.testDsc.Spec.Components.ModelRegistry.RegistriesNamespace),
+			jq.Match(`.status.phase == "%s"`, readyStatus),
+		)),
+	))
+}
+
+func (mr *ModelRegistryTestCtx) validateOperandsOwnerReferences(t *testing.T) {
+	g := mr.WithT(t)
+
+	g.Eventually(
+		mr.List(
+			gvk.Deployment,
+			client.InNamespace(mr.applicationsNamespace),
+			client.MatchingLabels{labels.ComponentManagedBy: componentsv1.ModelRegistryInstanceName},
+		),
+	).Should(And(
+		HaveLen(1),
+		HaveEach(
+			jq.Match(`.metadata.ownerReferences[0].kind == "%s"`, componentsv1.ModelRegistryKind),
+		),
+	))
+}
+
+func (mr *ModelRegistryTestCtx) validateUpdateModelRegistryOperandsResources(t *testing.T) {
+	g := mr.WithT(t)
+
+	appDeployments, err := mr.kubeClient.AppsV1().Deployments(mr.applicationsNamespace).List(
+		mr.ctx,
+		metav1.ListOptions{
+			LabelSelector: labels.ComponentManagedBy + "=" + componentsv1.ModelRegistryInstanceName,
+		},
+	)
+
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(appDeployments.Items).To(HaveLen(1))
+
+	const expectedReplica int32 = 2 // from 1 to 2
+
+	testDeployment := appDeployments.Items[0]
+	patchedReplica := &autoscalingv1.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testDeployment.Name,
+			Namespace: testDeployment.Namespace,
+		},
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: expectedReplica,
+		},
+		Status: autoscalingv1.ScaleStatus{},
+	}
+
+	updatedDep, err := mr.kubeClient.AppsV1().Deployments(mr.applicationsNamespace).UpdateScale(
+		mr.ctx,
+		testDeployment.Name,
+		patchedReplica,
+		metav1.UpdateOptions{},
+	)
+
+	g.Expect(err).ShouldNot(HaveOccurred())
+	g.Expect(updatedDep.Spec.Replicas).Should(Equal(patchedReplica.Spec.Replicas))
+
+	g.Eventually(
+		mr.List(
+			gvk.Deployment,
+			client.InNamespace(mr.applicationsNamespace),
+			client.MatchingLabels{labels.ComponentManagedBy: componentsv1.ModelRegistryInstanceName},
+		),
+	).Should(And(
+		HaveLen(1),
+		HaveEach(
+			jq.Match(`.spec.replicas == %d`, expectedReplica),
+		),
+	))
+
+	g.Consistently(
+		mr.List(
+			gvk.Deployment,
+			client.InNamespace(mr.applicationsNamespace),
+			client.MatchingLabels{labels.ComponentManagedBy: componentsv1.ModelRegistryInstanceName},
+		),
+	).WithTimeout(30 * time.Second).WithPolling(1 * time.Second).Should(And(
+		HaveLen(1),
+		HaveEach(
+			jq.Match(`.spec.replicas == %d`, expectedReplica),
+		),
+	))
+}
+
+func (mr *ModelRegistryTestCtx) validateModelRegistryCert(t *testing.T) {
```

Comment:
not related.
if we want to test cert in e2e, will be diffcult to run e2e locally without OCP

Reply:
not sure those those e2e tests are supposed to run outside of OCP, are they ?

Reply:
i dont think so, at least not for now to run e2e locally.